src/main/java/me/itzg/helpers/curseforge/CurseForgeInstaller.java

```diff
@@ -972,7 +975,13 @@ private void prepareModLoader(String id, String minecraftVersion) {
         }
 
         final String provider = parts[0];
-        final String loaderVersion = parts.length == 2 ? parts[1] : parts[2];
+        String loaderVersion = parts.length == 2 ? parts[1] : parts[2];
+
+        // Override with custom versions if provided
+        if (customModLoaderVersion != null) {
+            log.info("Overriding mod loader version from {} to {}", loaderVersion, customModLoaderVersion);
+            loaderVersion = customModLoaderVersion;
+        }
```

Comment:
I feel pretty strongly about `final`s, so changing something to be un-final is almost a sin 😄 

Now that the logic simpler, do it this way

```suggestion
        final String loaderVersion;
        // Override with custom versions if provided
        if (customModLoaderVersion != null) {
            log.info("Overriding mod loader version from {} to {}", loaderVersion, customModLoaderVersion);
            loaderVersion = customModLoaderVersion;
        }
        else {
            loaderVersion = parts.length == 2 ? parts[1] : parts[2];
        }
```